shell.go

```diff
@@ -72,16 +73,39 @@ func NewShell(url string) *Shell {
 	return NewShellWithClient(url, c)
 }
 
-func NewShellWithClient(url string, c *gohttp.Client) *Shell {
-	if a, err := ma.NewMultiaddr(url); err == nil {
-		_, host, err := manet.DialArgs(a)
-		if err == nil {
-			url = host
+func NewShellWithClient(url string, client *gohttp.Client) *Shell {
+	var sh Shell
+	sh.httpcli = *client
+
+	if maddr, err := ma.NewMultiaddr(url); err == nil {
+		if network, host, err := manet.DialArgs(maddr); err == nil {
+			if network == "unix" {
+				url = network
+				if tpt, ok := sh.httpcli.Transport.(*gohttp.Transport); ok {
+					var tptCopy *gohttp.Transport
+					if tpt == nil {
```

Comment:
If the transport is nil, will it even hit this case? Won't `sh.httpcli.Transport.(*gohttp.Transport)` fail?

Reply:
I honestly didn't think it was case, sorry for that..

Reply:
Np. I had to look it up myself to make sure.